src/functions.php

```diff
@@ -1,5 +1,83 @@
 <?php
 
+/**
+ * @param mixed $con
+ * @param integer $userId
+ * @param integer $projectId
+ * @param string $deadline
+ * @return array
+ */
+function getUserTasksInTimeInterval ($con, int $userId, string $deadline, ?int $projectId): array
```

Comment:
Второй параметр лучше сделать null по умолчанию и перенести в конец и проверь если не null, тогда сравнивать с датой.
```suggestion
function getUserTasksInTimeInterval (object $con, int $userId, ?int $projectId, ?string $deadline = null): array
```